src/Language/Futhark/Prop.hs

```diff
@@ -879,6 +879,49 @@ intrinsics =
                  ]
                  $ Array () Unique t_a (rank 1)
              ),
+             ( "scatter_2d",
+               IntrinsicPolyFun
+                 [tp_a]
+                 [ uarr_2d_a,
+                   Array
+                     ()
+                     Nonunique
+                     ( Record
+                         ( M.fromList $
+                             zip
+                               tupleFieldNames
+                               [ Scalar $ Prim $ Signed Int64,
+                                 Scalar $ Prim $ Signed Int64
+                               ]
+                         )
+                     )
+                     (rank 1),
+                   Array () Nonunique t_a (rank 1)
+                 ]
+                 uarr_2d_a
+             ),
+             ( "scatter_3d",
+               IntrinsicPolyFun
+                 [tp_a]
+                 [ uarr_3d_a,
+                   Array
+                     ()
+                     Nonunique
+                     ( Record
+                         ( M.fromList $
+                             zip
+                               tupleFieldNames
+                               [ Scalar $ Prim $ Signed Int64,
+                                 Scalar $ Prim $ Signed Int64,
+                                 Scalar $ Prim $ Signed Int64
+                               ]
+                         )
+                     )
+                     (rank 1),
+                   Array () Nonunique t_a (rank 1)
+                 ]
+                 uarr_3d_a
+             ),
```

Comment:
This is many more lines than the other builtins.  Could you use some local definitions to compress it a bit?

Reply:
Good point! I'll rewrite it.

Reply:
This is overkill and an unnecessary indirection.  All you really need is a helper function for creating types of  tuples of integers, then the rest of the type can be inline.

Reply:
How about now?

Reply:
The tuple array parameter type will fit on one line:

```
                   Array ()	Unique (tupInt64 3) (rank 1)
```

This will make the definition as concise as the others.

Reply:
Done!